yarn-project/p2p/src/client/p2p_client.ts

```diff
@@ -56,7 +57,7 @@ export interface P2PSyncState {
 /**
  * Interface of a P2P client.
  **/
-export interface P2P extends P2PApi {
+export type P2P<T extends P2PClientType> = P2PApi<T> & {
```

Comment:
FYI you can set defaults for generic arguments, which would have meant a lot less code changes around. No need to make this change now though!

```suggestion
export type P2P<T extends P2PClientType = P2PClientType.Full> = P2PApi<T> & {
```

Reply:
🤦

Reply:
I think I will make it for easier maintaining